### PR TITLE
Android build improvements

### LINKED ISSUE #13938: Android build should check well known locations for NDK/cmake/ninja
##### System information (version)
- OpenCV => latest

##### Detailed description
In order to ease the pain for someone getting started on Android the following build improvements would be beneficial:
* check well known locations under the Android SDK directory for NDK/cmake/ninja
* if ccache is not found then disable ccache in the build
* better log output to indicate the cause of build failures

I have a PR prepared to address this issue

### PR DESCRIPTION
**Pull request by komakai**
### This pullrequest changes platform/android/build_sdk.py

* look for NDK in "ndk-bundle" directory under the SDK directory if location not specified
* look for cmake and ninja in "cmake" directory under the SDK directory if not found in the path
* disable use of ccache if not found in the path

### CODE REVIEW

File: platforms/android/build_sdk.py [hunk 1]
```diff
@@ -48,6 +48,53 @@ def check_dir(d, create=False, clean=False):
...
+        if os.curdir not in path:
+            path.insert(0, os.curdir)
+
+            # need to consider PATHEXT on Windows.
+            pathext = os.environ.get("PATHEXT", "").split(os.pathsep)
+            if any(cmd.lower().endswith(ext.lower()) for ext in pathext):
+                files = [cmd]
+            else:
+                files = [cmd + ext for ext in pathext]
+    else:
```
Comment (alalek) [inline platforms/android/build_sdk.py]: Code is not complete.

I believe we can use something like this instead:
```
def check_executable(cmd)
    try:
        retcode = subprocess.call(cmd, shell=shell)
        if retcode < 0:
            return False
        return True
    except:
        return False

if not check_executable(['ninja', "--version"])
    ... fallback for manual searching ...
if not check_executable(['cmake', "--version"])
    ...
```

**Comment by komakai** [comment]
> What about adding script parameter with force using of CMake/Ninja from Android SDK?

Might be a good idea - but probably something for a separate PR

File: platforms/android/build_sdk.py (hunk 1 above)
Comment (komakai) [reply platforms/android/build_sdk.py]: Will add something like this and push a new commit

Comment (komakai) [reply platforms/android/build_sdk.py]: Pushed a new commit to address this.
I tested the changes on Windows and Mac and they seem to work well

**Pull request merged**